react-native/host/src/provider/MobileWalletProtocolProvider.tsx

```diff
@@ -34,7 +34,7 @@ type MWPHostContextType = {
   handleRequestUrl: (url: string) => Promise<boolean>;
   fetchClientAppMetadata: () => Promise<AppMetadata | null>;
   isClientAppVerified: () => Promise<boolean>;
-  approveHandshake: (action: HandshakeAction, metadata: AppMetadata) => Promise<void>;
+  approveHandshake: (metadata: AppMetadata | null) => Promise<void>;
```

Comment:
not a js/ts experts, but afaik, we'd better use `undefined` unless `null` has a specific meaning on the context. https://github.com/Microsoft/TypeScript/wiki/Coding-guidelines#null-and-undefined  

how about updating like
```suggestion
  approveHandshake: (metadata: AppMetadata | undefined) => Promise<void>;
```
and line 35 to return the same type as well.

Reply:
@bangtoven, I wanted it to be intentional on the consumer's side not to pass metadata in. With undefined, it's easy to forget to pass in the param and just call `approveHandshake()`. The `fetchAppMetadata` function also returns a `AppMetadata | null`, so this works out well.